src/architecture/configuration.md

```diff
@@ -6,118 +6,109 @@
-configuration is passed to all services deployed in the blockchain.
-The configuration for a service is stored in the `services_configs` subtree
-of the overall configuration
-under a separate key equal to
-[the name of the service](services.md#service-identifiers).
+[Exonum services](services.md) may have their own configuration settings.
 
 !!! note "Example"
     The configuration settings of
     [the anchoring service](../advanced/bitcoin-anchoring.md)
     include the parameters of the RPC connection to
     a Bitcoin Core node as well as a Bitcoin address used for anchoring.
-    These parameters are stored in the `services_configs.btc_anchoring`
-    section of the overall configuration.
 
 ## Configuration Parameters
 
-System configuration contains 2 types of configuration parameters:
+System configuration falls into 2 parts:
 
 - **Global parameters** must be identical for all full nodes in the network. The
   global parameters are saved in the blockchain when the genesis block is
-  created
-- **Local parameters** may differ for each node
+  created.
+- **Local parameters** may differ for each node. Local parameters for a node
+  are stored in the [TOML][toml] format. A path to the configuration file
+  is specified on the node start up.
 
 This categorization holds for both core and service parameters.
+The following table shows three possible parameter categories.
+
+| Scope   | Core | Anchoring service |
+|---------|------|-------------------|
+| Global  | Validators’ public keys | Anchoring address |
+| Local   | Validator’s private key | (N/A) |
+
+Note that there is no unified mechanism for local service configuration.
+If necessary, local configuration may be injected into the service
+using a parameterized service artifact, or by separating
+the config-dependent part into a component executing separately
+and communicating with the service using a secure communication
+channel (e.g., [private API](../glossary.md#private-api)).
+
+!!! example
+    [The time oracle](../advanced/time.md) illustrates the first approach
+    (the service factory accepts a time provider),
+    and [the anchoring service](../advanced/bitcoin-anchoring.md)
+    the second one (the part communicating with the Bitcoin node is
+    separated into an independent executable).
+
+Global configuration may be changed using [the supervisor service](../advanced/supervisor.md),
+and local configuration [by editing the configuration file](#changing-configuration).
+When the blockchain is being created, the initial global configuration
+is supplied from a file, too, since the blockchain is absent at this point.
+
+## Genesis Configuration
+
+**Genesis configuration** is the global configuration necessary to start
+a new node. While nodes update the global configuration using
+the supervisor service, a node need to get the genesis configuration from
+a trusted source. Indeed, at the point the genesis config is used, the node
```

Comment:
```suggestion
a trusted source. Indeed, at this point the genesis config is used, the node
```

Reply:
This is not correct, IMHO. We are talking about **the** point *when* config will be used.